webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,254 @@
+    const {formatMessage} = useIntl();
+
+    const originalChannel = useSelector((state: GlobalState) => getChannel(state, {id: post.channel_id}));
+    const currentTeam = useSelector(getCurrentTeam);
+
+    const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    const [bodyHeight, setBodyHeight] = useState<number>(0);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [postError, setPostError] = useState<React.ReactNode>(null);
+    const [selectedChannel, setSelectedChannel] = useState<ChannelOption>();
+    const [isButtonClicked, setIsButtonClicked] = useState<boolean>(false);
+
+    const bodyRef = useRef<HTMLDivElement>();
+
+    const measuredRef = useCallback((node) => {
+        if (node !== null) {
+            bodyRef.current = node;
+            setBodyHeight(node.getBoundingClientRect().height);
+        }
+    }, []);
+
+    const onHide = useCallback(() => {
+        onExited?.();
+    }, [onExited]);
+
+    const handleChannelSelect = useCallback((channel: ValueType<ChannelOption>) => {
+        if (Array.isArray(channel)) {
+            setSelectedChannel(channel[0]);
+            return;
+        }
+        setSelectedChannel(channel as ChannelOption);
+    },
+    [],
+    );
+
+    const messagePreviewTitle = formatMessage({
+        id: 'move_thread_modal.preview.title',
+        defaultMessage: 'Message preview',
+    });
+
+    const previewMetaData: PostPreviewMetadata = useMemo(() => ({
+        post,
+        post_id: post.id,
+        team_name: currentTeam.name,
+        channel_display_name: originalChannel.display_name,
+        channel_type: originalChannel.type,
+        channel_id: originalChannel.id,
+    }), [post, currentTeam.name, originalChannel.display_name, originalChannel.type, originalChannel.id]);
+
+    const notificationText = formatMessage({
+        id: 'move_thread_modal.notification.dm_or_gm',
+        defaultMessage: 'Moving this thread changes who has access',
+    });
+
+    const notification = (
+        <NotificationBox
+            variant={'info'}
+            text={notificationText}
+            id={'move_thread'}
+        />
+    );
+
+    const handlePostError = useCallback((error: ClientError) => {
+        setIsButtonClicked(false);
+        setPostError(error.message);
+        setHasError(true);
+
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+
+        timeoutRef.current = setTimeout(() => {
+            setHasError(false);
+            timeoutRef.current = null;
+        }, Constants.ANIMATION_TIMEOUT);
+    }, [setIsButtonClicked, setPostError, setHasError]);
+
+    const handleSubmit = useCallback(async () => {
+        setIsButtonClicked(true);
+        if (!selectedChannel) {
+            setIsButtonClicked(false);
+            return;
+        }
+
+        const channel = selectedChannel.details;
+
+        let result = await actions.moveThread(post.root_id || post.id, channel.id);
```

Comment:
This (and other awaits) may take an arbitrary amount of time to be handled. The user may have closed the modal and this may be unmounted. Why is this a problem?

- You may be setting the state for a unmounted component
- You may be doing side effects that no longer make sense (like the switch to channel done later)

Not 100% sure if any of that is a problem here, but would be great to verify.